test_themes/tests/test_crawl.py

```diff
@@ -51,3 +147,96 @@ def test_02_homepage_tour_every_theme(self):
         websites_themes = Website.get_test_themes_websites()
         for website in websites_themes:
             self.start_tour(f"/web?fw={website.id}", 'homepage', login='admin')
+
+    def test_template_names(self):
+        websites_themes = self.env['website'].get_test_themes_websites()
+        for website in websites_themes:
+            for view in self.env['ir.ui.view'].search([
+                ('key', 'like', f'{website.theme_id.name}.new_page_template%_s_'),
+            ]):
+                self.assertEqual(view.mode, 'extension', "Theme's new page template customization %r should never be primary" % view.key)
+                name = view.key.split('.')[1]
+                parent_name = view.inherit_id.key.split('.')[1]
+                self.assertEqual(name, parent_name, "Theme's new page template customization %r should use the same name as their parent %r" % (view.key, view.inherit_id.key))
+
+    def test_render_templates(self):
+        errors = []
+        view_count = 0
+        websites_themes = self.env['website'].get_test_themes_websites()
+        for website in websites_themes:
+            with MockRequest(self.env, website=website):
+                views = self.env['ir.ui.view'].search([
+                    '|', '|',
+                    ('key', 'like', f'{website.theme_id.name}.s_'),
+                    ('key', 'like', f'{website.theme_id.name}.configurator'),
+                    ('key', 'like', f'{website.theme_id.name}.new_page'),
+                ])
+                for view in views:
+                    try:
+                        self.env['ir.qweb']._render(view.id)
+                    except Exception:
+                        errors.append("View %s cannot be rendered" % view.key)
+                        raise
+                view_count += len(views)
+        _logger.info("Tested %s views", view_count)
```

Comment:
I now use a set of ids in `test_render_templates`, but I kept a count in `test_render_applied_templates` - since it is always the same views that are rendered for each theme.